hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/TableCommand.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.cli.commands;
+
+import com.uber.hoodie.cli.HoodieCLI;
+import com.uber.hoodie.cli.HoodiePrintHelper;
+import com.uber.hoodie.cli.TableHeader;
+import com.uber.hoodie.common.model.HoodieTableType;
+import com.uber.hoodie.common.table.HoodieTableMetaClient;
+import com.uber.hoodie.exception.DatasetNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.stereotype.Component;
+
+/**
+ * Commands to create and describe hoodie tables.
+ */
+@Component
+public class TableCommand implements CommandMarker {
+
+  @CliAvailabilityIndicator({"table desc"})
+  public boolean isAvailable() {
+    return HoodieCLI.tableMetadata != null;
+  }
+
+  /**
+   * Create a Hoodie Table if it does not exist
+   * @param path Base Path
+   * @param name Hoodie Table Name
+   * @param tableTypeStr Hoodie Table Type
+   * @param payloadClass Payload Class
+   * @return
+   * @throws IOException
+   */
+  @CliCommand(value = "table create", help = "Create a hoodie table if not present")
+  public String createTable(
+      @CliOption(key = {"path"}, mandatory = true, help = "Base Path of the dataset") final String path,
+      @CliOption(key = {"tableName"}, mandatory = true, help = "Hoodie Table Name") final String name,
+      @CliOption(key = {"tableType"}, unspecifiedDefaultValue = "COPY_ON_WRITE",
+      help = "Hoodie Table Type. Must be one of : COPY_ON_WRITE or MERGE_ON_READ") final String tableTypeStr,
+      @CliOption(key = {"payloadClass"}, unspecifiedDefaultValue = "com.uber.hoodie.common.model.HoodieAvroPayload",
+      help = "Payload Class") final String payloadClass) throws IOException {
+
+    boolean initialized = HoodieCLI.initConf();
```

Comment:
this command may be called multiple times, in that scenario we set exiting = false everytime, should we first check for existence ?

Reply:
This logic also takes care of the case when FS is already initialized. As this command is special (This command can be invoked with/without connect), this command initializes FS if not set. Otherwise, it is a no-op.

Reply:
can we reuse the DatasetsCommand?  Its what we use to connect and operate on? or we bring that here (and update docs)

Reply:
@vinothchandar : Makes sense. I have updated the diff moving the change to Datasets command. The related document changes is in the PR : https://github.com/uber/hudi/pull/445